doc/ADDB_Monitoring.rst

```diff
@@ -0,0 +1,281 @@
+
+ Execution of monitor
+======================
+
+Partial execution i.e. top half is executed for all the addb records during monitor invocation. This just updates the monitor specific summary data or posts an ast if an exception had been encountered.
+
+There is a periodic posting of these addb summary records and this is done by the locality thread on the m0d. This thread also runs the ASTs that are posted for the exception summary records. 
+
+The bottom half i.e. AST part would be run by a dedicated thread & would be synchronized among the various others threads that would run monitors with a sm (state machine) group lock.
+
+ Conformance
+===============
+
+- [i.addb.monitor.add] An API is made available for this. 
+
+- [i.addb.monitor.remove] An API is made available for this. 
+
+- [i.addb.monitor.generate-summary-addb-records] Monitor’s am_watch() function will do this. 
+
+- [r.addb.monitor.deliver-addb-record-to-stats-service] Addition to current ADDB mechanism is to be done to differentiate between summary stats records generated by monitors and other addb records & send these summary records to stats service. 
+
+- [r.addb.monitor.deliver-addb-exception-records-to-local-HA] Monitor’s am_watch() function will do this. 
+
+- [r.addb.monitor.deliver-addb-record-to-addb-service] This makes use of current implementation. 
+
+- [r.addb.monitor.nesting] Monitors generate addb records which themselves can be monitored. 
+
+- [r.addb.stats-service.state] Implementation of stats service handles this. 
+
+- [r.addb.stats-service.query] Implementation of stats service handles this. 
+
+- [r.addb.stats-service.single-instance] Implementation of stats service handles this.
+
+ Dependencies
+==============
+
+- [r.addb.retention] ADDB monitor generates addb records. 
+
+- [r.addb.retention.storage] ADDB monitor generates addb records. 
+
+- [r.addb.timings] ADDB monitor may need to calculate processing rate statistics. 
+
+- [r.addb.filtering] ADDB monitor needs information from addb records. 
+
+- [r.addb.record.type.datapoint] ADDB monitor can generate datapoint addb records. 
+
+- [r.addb.record.type.counter] ADDB monitor can generate counter addb records. 
+
+- [r.addb.record.type.event] ADDB monitor can generate event addb record 
+
+- [r.addb.record.type.counter.statistics] ADDB monitor needs to do statistics reporting. 
+
+- [r.addb.record.definition] ADDB monitor can define new addb record. 
+
+- [r.addb.record.definition.extensible] 
+
+- [r.addb.post] ADDB monitor can post addb records. 
+
+- [r.addb.post.non-blocking] Decrease performance impact of ADDB monitoring.
+
+ Use Cases
```

Comment:
"Use Cases" has the same level with "Functional Specification" and "Logical Specification".

Reply:
Incorporated.